### ISSUE TITLE
_createSrcFileCommit doesnt respect branch flag

### ISSUE THREAD

**Issue by atanner27** [MunifTanjim/node-bitbucket#8]
https://bitbucketjs.netlify.com/#api-repositories-repositories_createSrcFileCommit

Using this with branch: 'example' results in commits directly against master, regardless of what is supplied

**Comment by MunifTanjim** [comment]
```js
bitbucket.repositories
  .createSrcFileCommit({
    username: 'test_user',
    repo_slug: 'test_repo',
    _body: { data: 'test_data' },
    branch: 'test_branch'
  })
```

This produces the following request:
```
https://api.bitbucket.org/2.0/repositories/test_user/test_repo/src?branch=test_branch
{
  method: 'POST',
  headers: {
    accept: 'application/json',
    'user-agent': 'NodeBitBucket/1.3.1',
    'content-type': 'application/x-www-form-urlencoded'
  },
  body: 'data=test_data'
}
```

Which is the expected request according to: [Doc: POST /2.0/repositories/{username}/{repo_slug}/src](https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D/src#post)

So, either something's wrong with Bitbucket API, or in the Bitbucket API's documentation.

If you can find out what's wrong feel free to share.

**Comment by atanner27** [comment]
I have the request working via axios in the following method

  let url = 'https://api.bitbucket.org/2.0/repositories/' + teamName + '/' + repo_slug + '/src';
const data = {
  'package.json': file,
  'message': 'hello world',
  'author': 'John Doe <john.doe@example.com>',
  'branch': branch
  };
const options = {
  method: 'POST',
  auth: {
    username: 'username',
    password: 'password'
  },
  headers: { 'content-type': 'application/x-www-form-urlencoded' },
  data: qs.stringify(data),
  url,
};


But this is using the form encoding instead of the other options

**Comment by MunifTanjim** [comment]
So, the API documentation is wrong. We have to put everything in the body. I'll add a fix.

**Comment by MunifTanjim** [comment]
@atanner27 Can you please confirm that 0215e52 fixes your problem?

If it's working alright for you, I'll publish a patched version.

**Comment by MunifTanjim** [comment]
fixed on [bitbucket@1.4.0](https://www.npmjs.com/package/bitbucket/v/1.4.0)
